src/index.ts

```diff
@@ -333,7 +333,7 @@ export class MvcValidationProviders {
             return false;
         }
 
-        let r = /^\+?[0-9\-\s]+$/;
+        let r = element.pattern ? new RegExp(element.pattern) : /^\+?[0-9\-\s]+$/;
```

Comment:
Maybe it should be:

```
let pattern = element.getAttribute(‘pattern’);
let r = pattern ? new RegExp(pattern) : /^\+?[0-9\-\s]+$/;
```

Reply:
> But why is it trying to get the pattern attribute from a select?
>
> > Property 'pattern' does not exist on type 'HTMLSelectElement'.

Because `element: ValidatableElement` and only `HTMLInputElement` has `pattern`:
https://github.com/haacked/aspnet-client-validation/blob/061c242280ef535bbaadd7e5384695aba119eb3b/src/index.ts#L32

The `getAttribute` solution works, but you can make TS happy with either:
```suggestion
        let r = 'pattern' in element && element.pattern ? new RegExp(element.pattern) : /^\+?[0-9\-\s]+$/;
```
or
```suggestion
        let r = element instanceof HTMLInputElement && element.pattern ? new RegExp(element.pattern) : /^\+?[0-9\-\s]+$/;
```

While we're in the area, it could be handy to assign/use the default pattern as something like `static phonePattern` to allow overriding everywhere by assigning `MvcValidationProviders.phonePattern`.